src/unix/unix.c

```diff
@@ -114,27 +123,36 @@ define_closure_function(1, 1, context, default_fault_handler,
        support multiple processes, we may need to install current when
        resuming deferred processing. */
     process p = current->p;
-
-    if (frame[FRAME_VECTOR] == 14) {
-        u64 vaddr = fault_address(frame);
-        vmap vm = vmap_from_vaddr(p, vaddr);
-        if (vm == INVALID_ADDRESS) {
-            if (user) {
-                pf_debug("no vmap found for addr 0x%lx, rip 0x%lx", vaddr, frame[FRAME_RIP]);
-                deliver_fault_signal(SIGSEGV, current, vaddr, SEGV_MAPERR);
-
-                /* schedule this thread to either run signal handler or terminate */
-                schedule_frame(frame);
-                return 0;
-            } else {
-                rprintf("\nUnhandled page fault in kernel mode: ");
-                goto bug;
-            }
+    if (frame[FRAME_VECTOR] == 0) {
+      if (current_cpu()->state == cpu_user) {
+        deliver_fault_signal(SIGFPE, current, 0, FPE_INTDIV);
```

Comment:
You should pass `fault_address(frame)` as `vaddr` argument to deliver_fault_signal().
Also, do not change the indentation of the the existing code.